canonical-webteam/deployment-configs
# Configs for blog.ubuntu.com & blog.staging.ubuntu.com

And redirect insights.ubuntu.com and insights.staging.ubuntu.com to blog.ubuntu.com and blog.staging.ubuntu.com.

QA
--

Deploy blog.ubuntu.com, and the insights ingresses:

``` bash
./qa-deploy --staging blog.staging.ubuntu.com --production blog.ubuntu.com --tag 66deae3f88c8a5f251ac97a3adf8f81680139780-1525423606
./qa-deploy --production insights.ubuntu.com --staging insights.staging.ubuntu.com
```

Point all domains at minikube:

``` bash
echo "`minikube ip` blog.ubuntu.com blog.staging.ubuntu.com insights.ubuntu.com insights.staging.ubuntu.com" | sudo tee -a /etc/hosts
```

Watch the services come up:

``` bash
watch kubectl get all,ingress
```

When everything is up, check http://blog.ubuntu.com and http://blog.staging.ubuntu.com. Check http://insights.ubuntu.com and http://insights.staging.ubuntu.com redirect to blog.

Also check the redirects on the command-line:

``` bash
$ curl -sI -H 'Host: insights.ubuntu.com' `minikube ip` | grep Location
Location: https://blog.ubuntu.com/
$ curl -sI -H 'Host: insights.staging.ubuntu.com' `minikube ip` | grep Location
Location: https://blog.staging.ubuntu.com/
```

Now clean up your `/etc/hosts`:

``` bash
sudo sed -i '/blog.staging.ubuntu.com/d' /etc/hosts
```
